packages/react-native-oursky/src/RequirePermission.js

```diff
@@ -0,0 +1,131 @@
+// @flow
+import * as React from "react";
+import { Platform } from "react-native";
+import Permissions from "react-native-permissions";
+import OpenSettings from "react-native-open-settings";
+
+import Dialog from "./Dialog";
+import type { LayoutProps as DialogLayoutProps } from "./Dialog";
+
+const AndroidPermission = ["photo"];
+const IOSPermission = ["notification", "photo"];
+
+export type Props = {
+  permission: "notification" | "photo",
+
+  UndeterminedDialogProps: DialogLayoutProps,
+  DeniedDialogProps: DialogLayoutProps,
+
+  onAccept: () => void,
+  onReject: () => void,
+};
+
+type Status = "authorized" | "denied" | "restricted" | "undetermined";
+
+type State = {
+  showUndeterminedDialog: boolean,
+  showDeniedDialog: boolean,
+};
+
+export default class RequirePermission extends React.PureComponent<
+  Props,
+  State
+> {
+  state = {
+    showUndeterminedDialog: false,
+    showDeniedDialog: false,
+  };
+
+  componentDidMount() {
+    const validPermission = Platform.select({
+      ios: IOSPermission,
+      android: AndroidPermission,
+    });
+    const requestPermission = this.props.permission;
+
+    if (validPermission.includes(requestPermission)) {
+      this.requestPermission(requestPermission);
+    }
+  }
+
+  requestPermission = async (permission: string) => {
+    try {
+      const granted: Status = await Permissions.check(permission);
+      switch (granted) {
+        case "undetermined":
+          this.setState({
+            showUndeterminedDialog: true,
+          });
+          break;
+        case "authorized":
+          this.props.onAccept();
+          break;
+        case "denied":
+        case "restricted":
```

Comment:
The API is not parallel here. Can you tell me why you invoke onAccept when permission is authorized but not invoke onReject for denied or restricted case while you invoke onReject when exception is thrown?

Reply:
We will show denied dialog first, and invoke `onReject` after user pressed any button of the denied dialog.

Reply:
we don't need to show any dialog If permission is authorized, so just invoke onAccept.

Reply:
Noted